index.js

```diff
@@ -440,19 +486,43 @@ function collectObj (stream, cb) {
   function end () { cb(null, rows) }
 }
 
+// Object, (k, v -> v) -> Object
 function mapObj (obj, fn) {
   Object.keys(obj).forEach(function (key) {
     obj[key] = fn(key, obj[key])
   })
   return obj
 }
 
-function filterObj (obj, fn) {
-  var res = {}
-  Object.keys(obj).forEach(function (key) {
-    if (fn(key, obj[key])) {
-      res[key] = obj[key]
+// KdbPoint -> OsmVersion
+function kdbPointToVersion (pt) {
+  return pt.value.toString('hex')
+}
+
+// OsmDb, OsmVersion -> [OsmId]
+function getWayNodesOfDeletedDoc (osm, version, cb) {
+  getTypeOfDeletedDoc(osm, version, function (err, type) {
+    if (err) return cb(err)
+    if (type === 'way') {
+      osm.log.get(version, function (err, node) {
+        if (err) return cb(err)
+        return cb(null, node.value.refs)
+      })
+    } else {
+      cb(null, [])
     }
   })
-  return res
+}
+
+// OsmDb, OsmVersion -> String
+function getTypeOfDeletedDoc (osm, version, cb) {
```

Comment:
See my mod of `del()` so that we store `type` on the deleted doc, so we would not need this lookup. `type` should never change so this should be consistent (although we never actually check whether `type` changes).